bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/handler/MideaACHandler.java

```diff
@@ -0,0 +1,634 @@
+            } else if (channelUID.getId().equals(CHANNEL_SWING_MODE)) {
+                connectionManager.sendCommand(CommandHelper.handleSwingMode(command, lastresponse, config.version),
+                        callbackLambda);
+            } else if (channelUID.getId().equals(CHANNEL_SCREEN_DISPLAY)) {
+                connectionManager.sendCommand(CommandHelper.handleScreenDisplay(command, lastresponse), callbackLambda);
+            } else if (channelUID.getId().equals(CHANNEL_TEMPERATURE_UNIT)) {
+                connectionManager.sendCommand(CommandHelper.handleTempUnit(command, lastresponse), callbackLambda);
+            } else if (channelUID.getId().equals(CHANNEL_SLEEP_FUNCTION)) {
+                connectionManager.sendCommand(CommandHelper.handleSleepFunction(command, lastresponse), callbackLambda);
+            } else if (channelUID.getId().equals(CHANNEL_ON_TIMER)) {
+                connectionManager.sendCommand(CommandHelper.handleOnTimer(command, lastresponse), callbackLambda);
+            } else if (channelUID.getId().equals(CHANNEL_OFF_TIMER)) {
+                connectionManager.sendCommand(CommandHelper.handleOffTimer(command, lastresponse), callbackLambda);
+            } else if (channelUID.getId().equals(CHANNEL_MAXIMUM_HUMIDITY)) {
+                connectionManager.sendCommand(CommandHelper.handleMaximumHumidity(command, lastresponse),
+                        callbackLambda);
+            }
+        } catch (MideaConnectionException | MideaAuthenticationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (MideaException | IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    /**
+     * To initialize an AC Thing, first check if discovery on your LAN has
+     * been completed. In the discovery process the IP address, IP port, device ID
+     * and version are established. Next for V.3 devices, if the token and key are not
+     * known, the cloud needs to be contacted to get the token and key using either
+     * the default NetHome Plus cloud or your own cloud account with your password and
+     * email. LAN discovery DOES NOT include the token key retrieval needed to communicate
+     * with the AC. V2 devices bypass the cloud connection step because they use a simplier
+     * hard-coded encryption. Next the Connection Manager is established. Then a command
+     * is formed and sent to retrieve the AC capabilities if they have not been
+     * discovered. Capabilities are not returned in the initial LAN Discovery.
+     * Lastly the routine polling, token key update and Energy polling frequency are set.
+     * 
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(MideaACConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
```

Comment:
First check if the configuration is valid, before setting the state to unknown.

Reply:
Would the state be initializing (I found that was not possible to assign). I thought it needed to be something, but will remove and see.

Reply:
What if all config are good, then the thing is never set to anything but online? I thought it needed to be something while going through the initialization()